clang/runtime/dpct-rt/include/dpct/bindless_images.hpp

```diff
@@ -845,16 +1057,38 @@ inline void map_resources(int count, external_mem_wrapper **handles,
 /// \param [in] q The queue used to unmap the resource with.
 inline void unmap_resources(int count, external_mem_wrapper **handles,
                             queue_ptr q_ptr = &get_default_queue()) {
-  if (detail::check_duplicate_entries(count, handles) &&
-      "Duplicate handle entries found during resource unmapping!")
-    ;
+  if (detail::check_duplicate_entries(count, handles)) {
+    throw std::runtime_error(
+        "Duplicate handle entries found during resource unmapping!");
+  }
 
   for (int i = 0; i < count; i++) {
     handles[i]->unmap_resource(*q_ptr);
   }
 }
 #endif // _WIN32
 
+/// @brief Imports external memory into a SYCL external memory object.
+/// @param extMem Pointer to the SYCL external memory object to be initialized.
+/// @param memHandleDesc Pointer to the external memory handle description.
+/// @throws std::runtime_error If no valid external memory handle is provided.
+inline void
+import_external_memory(sycl::ext::oneapi::experimental::external_mem *extMem,
+                       external_mem_handle_desc *memHandleDesc) {
+  if (memHandleDesc->get_fd_handle() != -1) {
```

Comment:
Does it need to support Win32 resource on Linux or support Linux resource on Win32?

Reply:
We are not providing any cross support for Linux or Win32

Its just 
**Win32 handle** supported on Windows
and
**File Descriptor** supported on Linux

Reply:
So we don't need check fd handle on Win32, and don't need to check win32 handle on Linux.